app/controllers/topics_controller.rb

```diff
@@ -28,6 +30,9 @@ def destroy
   end
 
   def edit
+    unless @topic.user_id == current_user.id
+      render :show
+    end
```

Comment:
Again, this works, but it would be better to just scope the finding of the topic to the current user. Then if the current_user doesn't own the topic you return a 404, as if it didn't even exist.

```ruby
@topic = current_user.topics.find(params[:id])
render :show
```

...but this should be driven by a test: a request spec.